### PR TITLE
fix: version should be set by pyproject.toml

### PR DESCRIPTION
**Pull request by dbluhm**
`aries_cloudagent.version.__version__` was out of sync with the version listed in the pyproject.toml file. The pyproject.toml file should be the source of truth as it will be used when publishing using poetry. However, `__version__` is used in code for some operations. To ensure that the pyproject file and the `__version__` can't get out of sync, `aries_cloudagent.version` will now be read from the package metadata.

Update: I was previously trying to read from pyproject.toml. This was a naive approach that wouldn't have worked after installing ACA-Py as a pip package. The better solution is the one I've switched to, just reading the package metadata.

### CODE REVIEW

**Comment by swcurran** [comment]
Dang — I didn’t get the memo about that addition. I’ll update the “Publishing.md” doc to add that.  However, in that document, I did say **double check that there is not the version somewhere else** so I should have caught it…

🦤 


Bigger question — does this mean we need to do a 0.10.2 to get this properly published?

**Comment by dbluhm** [comment]
@swcurran No, 0.10.1 is valid since it was published before we merged the poetry PR. So no changes required for the release.

**Comment by swcurran** [comment]
Of course. Phewww…. I didn’t mess up :-)

**Review by swcurran — DISMISSED** [review]

**Review by usingtechnology — APPROVED** [review]
great catch and fix.

**Comment by dbluhm** [comment]
@swcurran @usingtechnology Sorry, fixing the conflicts on poetry.lock dismissed your reviews -- this should be ready to merge now though!

**Review by usingtechnology — DISMISSED** [review]

**Comment by dbluhm** [comment]
I need to resolve issues in integration tests. My path to the pyproject.toml is bad. This isn't urgent (until we start preparing for another release) so I'll mark this as a draft and we can get some of the other pending PRs merged first.

**Comment by dbluhm** [comment]
@usingtechnology @swcurran this is ready for another review.

**Review by swcurran — APPROVED** [review]

**Review by usingtechnology — APPROVED** [review]

**Pull request merged**
